src/App.tsx

```diff
@@ -14,11 +14,107 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
 
-export default App;
+interface State {
+  goods: string[];
+  isVisible: boolean;
+}
+
+export class App extends React.Component<Props, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isVisible: false,
+  };
+
+  show = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((firstGood, nextGood) => firstGood.localeCompare(nextGood)),
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: [...goodsFromServer],
```

Comment:
No need to make copy here.
```suggestion
      goods: goodsFromServer,
```

Reply:
Fixed